File: CodingDojo/games/engine/src/main/java/com/codenjoy/dojo/client/LengthToXY.java
package com.codenjoy.dojo.client;

import com.codenjoy.dojo.services.Point;
import com.codenjoy.dojo.services.PointImpl;

/**
 * User: oleksandr.baglai
 */
public class LengthToXY {
    private int size;

    public LengthToXY(int size) {
        this.size = size;
    }

    public Point getXY(int length) {
        if (length == -1) {
            return null;
        }
        return new PointImpl(length % size, length / size);
    }

    public int getLength(int x, int y) {
        return (y)* size + x;
    }
}

File: CodingDojo/games/icancode/src/main/dotNet/ICanCode.Api/LengthToXY.cs
namespace ICanCode.Api
{
    public class LengthToXY
    {
        public int Size;

        public LengthToXY(int size)
        {
            Size = size;
        }

        private int InversionY(int y)
        {
            return Size - 1 - y;
        }

        private int InversionX(int x)
        {
            return x;
        }

        public int GetLength(int x, int y)
        {
            int xx = InversionX(x);
            int yy = InversionY(y);
            return yy * Size + xx;
        }

        public Point GetXY(int length)
        {
            int x = InversionX(length % Size);
            int y = InversionY(length / Size);
            return new Point(x, y);
        }
    }
}
